lib/in_out/parser.py

```diff
@@ -0,0 +1,92 @@
+import logging
+import re
+
+from parser_regex import log_regex
+
+
+class Parser:
+    def __init__(self, client):
+
+        # self.nicks maintains a list of current user nicks present \
+        # on channels. We will use this heuristic to see nicks mention \
+        # in a message. User mention can happen in following ways: \
+        # <rohit> bhuvan: or <rohit> bhuvan, kausam, or
+        # <rohit> nice work kausam.
+
+        self.nicks = []
+        self.regexes = log_regex[client]
+        # setup logging
+        logging.basicConfig(filename='logger_out.log', level=logging.DEBUG)
+
+    def data_extractor(self, matchObj, groups):
+        data = {}
+        for category_name, group_no in groups.items():
+            data[category_name] = matchObj.group(group_no)
+
+        return data
+
+    def is_nick_change_event(self, line):
+
+        return re.match(self.regexes['nick_change']['regex'], line)
+
+    def parse_nick_change(self, log_line):
```

Comment:
Please solve the duplication issue in the next refactoring.

Reply:
done